grpc-server.go

```diff
@@ -621,3 +624,90 @@ func (multi *MultiEpoch) GetBlockTime(ctx context.Context, params *old_faithful_
 		BlockTime: int64(blocktime),
 	}, nil
 }
+
+func (multi *MultiEpoch) StreamBlocks(params *old_faithful_grpc.StreamBlocksRequest, ser old_faithful_grpc.OldFaithful_StreamBlocksServer) error {
+	ctx := ser.Context()
+
+	startSlot := params.StartSlot
+	endSlot := startSlot + maxSlotsToStream
+
+	if params.EndSlot != nil {
+		endSlot = *params.EndSlot
+	}
+
+	filterFunc := func(block *old_faithful_grpc.BlockResponse) bool {
+		if params.Filter == nil || len(params.Filter.AccountInclude) == 0 {
+			return true
+		}
+
+		return blockContainsAccounts(block, params.Filter.AccountInclude)
+	}
+
+	for slot := startSlot; slot <= endSlot; slot++ {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+		}
+
+		block, err := multi.GetBlock(ctx, &old_faithful_grpc.BlockRequest{Slot: slot})
+		if err != nil {
+			if status.Code(err) == codes.NotFound {
+				continue // is this the right thing to do?
+			}
+			return err
+		}
+
+		if filterFunc(block) {
+			if err := ser.Send(block); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func blockContainsAccounts(block *old_faithful_grpc.BlockResponse, accounts []string) bool {
+	accountSet := make(map[string]struct{}, len(accounts))
+	for _, acc := range accounts {
+		accountSet[acc] = struct{}{}
+	}
+
+	for _, tx := range block.Transactions {
+		decoded, err := iplddecoders.DecodeTransaction(tx.Transaction)
+		if err != nil {
+			klog.Warningf("Failed to decode transaction: %w", err)
+			continue // skip if there's error decoding
+		}
+		solTx, err := decoded.GetSolanaTransaction()
+		if err != nil {
+			klog.Warningf("Failed to get sol transaction: %w", err)
+			continue
+		}
+
+		for _, acc := range solTx.Message.AccountKeys {
+			if _, exists := accountSet[acc.String()]; exists {
+				return true
+			}
+		}
+
+		for _, inst := range solTx.Message.Instructions {
```

Comment:
this seems to be redundant

Reply:
So no loop required over instructions?

Reply:
all the accounts are in the tx. message 

unless you consider also the accounts from address lookup tables, and in that case you need to get that from tx meta

Reply:
Let’s not support address table lookups for now but add it as a future potential feature  (add a ticket for it ).

Reply:
done!